### PR TITLE
dev-libs/ocl-icd: change the vendors dir when use prefix

### PR DESCRIPTION
**Pull request by littlewu2508**
Signed-off-by: YiyangWu <xgreenlandforwyy@gmail.com>

### CODE REVIEW

File: dev-libs/ocl-icd/ocl-icd-2.3.0.ebuild [hunk 1]
```diff
@@ -15,7 +15,7 @@ KEYWORDS="amd64 x86"
 
 # Does nothing now but by keeping it here we avoid having to have virtual/opencl
 # handle ebuilds both with and without this flag.
-IUSE="+khronos-headers"
+IUSE="+khronos-headers prefix"
```
Comment (heroxbd) [inline dev-libs/ocl-icd/ocl-icd-2.3.0.ebuild:18]: Do we need the `prefix.eclass`?

**Review by heroxbd — CHANGES_REQUESTED** [review]

File: dev-libs/ocl-icd/ocl-icd-2.3.0.ebuild (hunk 1 above)
Comment (littlewu2508) [reply dev-libs/ocl-icd/ocl-icd-2.3.0.ebuild:18]: No we don't. We only have to check whether `prefix` USE flag is on, and use `${EPREFIX}` variable is currently enough

Comment (heroxbd) [reply dev-libs/ocl-icd/ocl-icd-2.3.0.ebuild:18]: I don't think you need to append `prefix` to `IUSE` either.  It is covered by IUSE_IMPLICIT.

see `profiles/base/make.defaults:8` and PMS 11.1.1 for details.

Comment (littlewu2508) [reply dev-libs/ocl-icd/ocl-icd-2.3.0.ebuild]: You're right. Now I have removed that use flag.

File: dev-libs/ocl-icd/ocl-icd-2.3.1.ebuild [hunk 2]
```diff
@@ -33,7 +33,8 @@ src_prepare() {
 multilib_src_configure() {
 	# dev-util/opencl-headers ARE official Khronos Group headers, what this option
 	# does is disable the use of the bundled ones
-	ECONF_SOURCE="${S}" econf --enable-pthread-once --disable-official-khronos-headers
+	ECONF_SOURCE="${S}" econf --enable-pthread-once --disable-official-khronos-headers \
+		$(use_enable prefix custom-vendordir="${EPREFIX}/etc/OpenCL/vendors")
```
[the remark this answers is not in the document]
Comment (heroxbd) [reply dev-libs/ocl-icd/ocl-icd-2.3.1.ebuild:37]: Find the correct usage of `use_enable` from the documentation https://devmanual.gentoo.org/function-reference/query-functions/

[the remark this answers is not in the document]
Comment (littlewu2508) [reply dev-libs/ocl-icd/ocl-icd-2.3.1.ebuild]: Thanks! I misused `use_enabled`

[the remark this answers is not in the document]
Comment (littlewu2508) [reply dev-libs/ocl-icd/ocl-icd-2.3.1.ebuild]: I came to the simpler solution now.

[the remark this answers is not in the document]
Comment (heroxbd) [reply dev-libs/ocl-icd/ocl-icd-2.3.1.ebuild]: That works, `$(use_enable prefix custom-vendordir "${EPREFIX}/etc/OpenCL/vendors")` will also do.

**Comment by littlewu2508** [comment]
> That works, `$(use_enable prefix custom-vendordir "${EPREFIX}/etc/OpenCL/vendors")` will also do.

I think this will also cause calling --disable-custom-vendordir on vanilla gentoo?

**Comment by heroxbd** [comment]
littlewu2508 ***@***.***> writes:

> I think this will also cause calling --disable-custom-vendordir on
> vanilla gentoo?

`--disable-custom-vendordir` works. Only
`--disable-custom-vendordir=xxxx` fails.

**Pull request closed without merging**